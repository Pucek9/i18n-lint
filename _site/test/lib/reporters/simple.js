/**
 * hslint simple reporter tests
 *
 * Copyright (c) 2015 James Warwood
 * Licensed under the MIT license.
 */
/* global describe, it */
'use strict';

var expect = require('chai').expect;
var fs = require('fs');
var hooker = require('hooker');
var stripAnsi = require('strip-ansi');

var reporter = require('../../../lib/reporters/simple.js').reporter;

var data = require('../../fixtures/errors/1.js')();

describe('hslint reporters simple', function() {
  it('should format the output correctly', function(done) {
    var actual = '';

    hooker.hook(process.stdout, 'write', {
      pre: function(out) {
        actual += stripAnsi(out);

        return hooker.preempt();
      }
    });
    var expected = fs.readFileSync(
      __dirname + '/../../expected/reporters/simple.txt'
    ).toString();

    // Execute method under test
    reporter(data);

    expect(actual).to.equal(expected);
    hooker.unhook(process.stdout, 'write');

    done();
  });
});
