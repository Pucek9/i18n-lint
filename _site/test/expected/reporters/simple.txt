somefile.html: line 5, column 5, Hardcoded <p> tag
somefile.html: line 7, column 19, Hardcoded 'alt' attribute
